pkg/kf/commands/apps/push.go

```diff
@@ -31,21 +32,22 @@ import (
 	utils "github.com/google/kf/pkg/kf/internal/utils/cli"
 	"github.com/google/kf/pkg/kf/manifest"
 	servicebindings "github.com/google/kf/pkg/kf/service-bindings"
+	glob "github.com/google/kf/third_party/cf-cli-glob"
 	"github.com/spf13/cobra"
 	"knative.dev/pkg/ptr"
 )
 
 // SrcImageBuilder creates and uploads a container image that contains the
 // contents of the argument 'dir'.
 type SrcImageBuilder interface {
-	BuildSrcImage(dir, srcImage string) error
+	BuildSrcImage(dir, srcImage string, filter func(path string) (bool, error)) error
```

Comment:
the definition `func(path string) (bool, error)` should probably be its own type since it's repeated all over

Reply:
This would require updating kontext. This is its function signature. That being said, I think we should pull kontext into `./third_party/forked` (https://github.com/google/kf/issues/110#issuecomment-539121228) and then we can more easily make these types of changes.

Reply:
I think you can create a type alias locally though with:

```
type KontextFilter = func(path string) (bool, error)
```

Note the = sign rather than just `type X Y`

Reply:
Ah, good idea. I never think of aliases.